rmtest/disposableredis/__init__.py

```diff
@@ -126,18 +140,36 @@ def __exit__(self, exc_type, exc_val, exc_tb):
         if exc_val or self.errored:
             sys.stderr.write("Redis output: {}\n".format(self._get_output()))
 
-    def dump_and_reload(self):
+    def _wait_for_child(self):
+        # Wait until file is available
+        r = self.client()
+        while True:
+            info = r.info('persistence')
+            if info['aof_rewrite_scheduled'] or info['aof_rewrite_in_progress']:
+                time.sleep(0.1)
+            else:
+                break
+
+    def dump_and_reload(self, restart_process=False):
         """
         Dump the rdb and reload it, to test for serialization errors
         """
         conn = self.client()
-        conn.save()
-        self.dumped = True
-        try:
-            conn.execute_command('DEBUG', 'RELOAD')
-        except redis.RedisError as err:
-            self.errored = True
-            raise err
+
+        if restart_process:
```

Comment:
How does this work when spawning a new process in general? the entire disposableredis only makes sense in this context.